app/src/main/java/com/davidtomas/taskyapp/features/agenda/presentation/agenda/AgendaRoot.kt

```diff
@@ -34,6 +79,22 @@ fun AgendaRoot(
             }
         }
     }
+
+    if (showPermissionDialog)
+        PermissionDialog(
+            permissionTextProvider = NotificationPermissionTextProvider(),
+            isPermanentlyDeclined = !shouldShowRequestPermissionRationale(
+                activity,
+                Manifest.permission.POST_NOTIFICATIONS
+            ),
+            onDismiss = { showPermissionDialog = false },
+            onOkClick = {
+                showPermissionDialog = false
+                notificationPermissionResultLauncher.launch(Manifest.permission.POST_NOTIFICATIONS)
+            },
+            onGoToAppSettingsClick = { activity.openAppSettings() }
+        )
+
```

Comment:
This will not survive config changes since you just use normal remember instead of rememberSaveable